### PR TITLE
Pset_FaceExcavationCommon

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_FaceExcavationCommon

## Description: 

Common properties for face excavation elements

| Property                       | Datatype                    | Description                                                  |
| ------------------------------ | --------------------------- | ------------------------------------------------------------ |
| FaceExcavationType             | PEnum_FaceExcavationType    | Type of face excavation                                      |
| FaceExcavationTypeOther        | IfcLabel                    | A user defined underground face excavation type when FaceExcavationType = OTHER |
| IsDriveDirectedAlongChainage | IfcBoolean  | Specifies whether the tunnel drive direction agrees with the positive chainage direction (true) or if the tunnel drive direction is reversed (false). |
| TunnelInclination              | PEnum_TunnelInclination     | Specifies the tunnel inclination relative to the drive direction as specified by IsDriveDirectedAlongChainage. |
| DriveInclination               | Penum_DriveInclination      | Specifies the tunnel drive inclination. |
| DriveInclinationOther          | IfcLabel                    | Specifies a user defined inclination of tunnel drive when DriveInclination = USERDEFINED. |
| ExcavationFaceInclination  | PEnum_ExcavationFaceInclination  | Specifies the inclination of the tunnel face. |
| ExcavationFaceInclinationOther           | IfcLabel                    | A user defined face inclination when ExcavationFaceInclination = USERDEFINED. |
| MaxUnsupportedExcavationLength | IfcNonNegativeLengthMeasure | Maximum unsupported length of excavation at the tunnel face  |

## Enumeration: PEnum_FaceExcavationType

| Constant        | Description |
| --------------- | ----------- |
| FULLFACE        |             |
| STAGED          |             |
| PILOTBORE       |             |
| SHAFT           |             |
| CROSSCONNECTION |             |
| REAMING         |             |
| RAISEDRILL      |             |
| OTHER           |             |

## Enumeration: PEnum_TunnelInclination
Enumeration for describing the tunnel inclination (longitudinal slope).

| Constant | Description |
| -------- | ----------- |
| RISING   | Tunnel has a rising inclination along the drive direction. |
| FALLING  | Tunnel has a falling inclination along the drive direction. |

## Enumeration: PEnum_DriveInclination
Enumeration of possible tunnel drive inclinations.

| Constant   | Description |
| ---------- | ----------- |
| HORIZONTAL | Tunnel drive is horizontal (parallel to sea level). |
| VERTICAL   | Tunnel drive is vertical (along the gravitational axis). |
| INCLINED   | Tunnel drive is inclined along the longitudinal slope. |
| USERDEFINED  | **Existing constant**  |

## Enumeration: PEnum_ExcavationFaceInclination
Enumeration of possible tunnel face inclinations.

| Constant | Description |
| -------- | ----------- |
| VERTICAL | The tunnel face is vertically inclined (along the gravitational axis). |
| INCLINED | The tunnel face is inclined (perpendicular to an inclined plane). |
| USERDEFINED   | **Existing constant**  |

### CODE REVIEW

File: IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml]: Looking at the values I would propose the following change to this property:
- remove the enum
- change property type to `IfcBoolean`
- rename the property to `IsDriveDirectedAlongChainage`

File: IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/Documentation.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Inclination of tunnel drive
```
Comment (SergejMuhic) [inline IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/Documentation.md]:
```suggestion
Tunnel drive inclination.
```

File: IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml]: Relative to what?

File: IFC4x3/Properties/f/FaceExcavationType_2bpyAfLPD1KPvgab$9KTzT/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Type of face excavation
```
Comment (SergejMuhic) [inline IFC4x3/Properties/f/FaceExcavationType_2bpyAfLPD1KPvgab$9KTzT/Documentation.md]:
```suggestion
Face excavation type.
```

File: IFC4x3/Properties/f/FaceInclinationOther_0vb$DrG6X2cOgI1cvUqXBm/Documentation.md [hunk 5]
```diff
@@ -0,0 +1 @@
+A user defined face inclination when FaceInclination = OTHER
```
Comment (SergejMuhic) [inline IFC4x3/Properties/f/FaceInclinationOther_0vb$DrG6X2cOgI1cvUqXBm/Documentation.md]:
```suggestion
A user defined face inclination when FaceInclination = OTHER.
```

File: IFC4x3/Properties/f/FaceInclination_0KcNnRVov7v8l33uxXOfnq/Documentation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Inclination of face inclination.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/f/FaceInclination_0KcNnRVov7v8l33uxXOfnq/Documentation.md]: Inclination of face inclination? 😄 

Should this have been "Excavation face inclination."?

File: IFC4x3/Properties/t/TunnelInclination_2$VkG7ESn3rQ2WJSlegUBN/Documentation.md [hunk 7]
```diff
@@ -0,0 +1 @@
+Inclination of tunnel in relation to the DriveDirection.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/t/TunnelInclination_2$VkG7ESn3rQ2WJSlegUBN/Documentation.md]:
```suggestion
Tunnel inclination in relation to DriveDirection.
```

File: IFC4x3/Properties/t/TunnelInclination_2$VkG7ESn3rQ2WJSlegUBN/DocProperty.xml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/Properties/t/TunnelInclination_2$VkG7ESn3rQ2WJSlegUBN/DocProperty.xml]: This sounds a bit complicated. Geometric data described by enumerations. Albeit I have no idea what added value this brings, if it is a requirement, then it is a requirement. 😄

**Review by SergejMuhic — COMMENTED** [review]
Property constants and property enumerations would be well served with descriptions.

File: IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml:1]: This comment should be answered by @JontyDunn

File: IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml]: > Relative to what?

This comment should be answered by @JontyDunn

File: IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml]: > Looking at the values I would propose the following change to this property:
> 
> * remove the enum
> * change property type to `IfcBoolean`
> * rename the property to `IsDriveDirectedAlongChainage`

This comment should be answered by @JontyDunn

File: IFC4x3/Properties/f/FaceInclination_0KcNnRVov7v8l33uxXOfnq/Documentation.md (hunk 6 above)
Comment (larswik) [reply IFC4x3/Properties/f/FaceInclination_0KcNnRVov7v8l33uxXOfnq/Documentation.md]: This comment should be answered by @JontyDunn

**Comment by larswik** [comment]
Ping @JontyDunn, @hessf-ilf (enum descriptions)

**Comment by larswik** [comment]
> Property constants and property enumerations would be well served with descriptions.

Ping @JontyDunn and @hessf-ilf. Tried to achieve some after meeting with @SergejMuhic and @Michel-Rives. These needs to be checked. For FaceExcavationType constant descriptions are still missing.

File: IFC4x3/Properties/t/TunnelInclination_2$VkG7ESn3rQ2WJSlegUBN/DocProperty.xml (hunk 8 above)
Comment (larswik) [reply IFC4x3/Properties/t/TunnelInclination_2$VkG7ESn3rQ2WJSlegUBN/DocProperty.xml]: Agreed to keep.

File: IFC4x3/Properties/f/FaceInclination_0KcNnRVov7v8l33uxXOfnq/Documentation.md (hunk 6 above)
Comment (larswik) [reply IFC4x3/Properties/f/FaceInclination_0KcNnRVov7v8l33uxXOfnq/Documentation.md]: Name changed to ExcavationFaceInclination (including PEnum_ExcavationFaceInclination).

File: IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml]: Added description. Check!

File: IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/d/DriveDirection_2PE$H34I5DPu1AAA2NFe$2/DocProperty.xml]: Property changed to IsDriveDirectedAlongChainage : IfcBoolean.

File: IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/d/DriveInclination_3wCXCi1srD59Rj89Iug8_e/DocProperty.xml]: Tried to improve description.

File: IFC4x3/Properties/e/ExcavationFaceInclinationOther_0vb$DrG6X2cOgI1cvUqXBm/Documentation.md [hunk 9]
```diff
@@ -0,0 +1 @@
+A user defined face inclination when ExcavationFaceInclination = USERDEFINED.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/ExcavationFaceInclinationOther_0vb$DrG6X2cOgI1cvUqXBm/Documentation.md]: OTHER or USERDEFINED?

File: IFC4x3/PropertyConstants/c/CROSSCONNECTION_0kL1OOai9AIeHS7Dm9jfFu.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CROSSCONNECTION_0kL1OOai9AIeHS7Dm9jfFu" Name="CROSSCONNECTION" UniqueId="2e541618-92c2-4a4a-845c-1cdc09b693f8" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/c/CROSSCONNECTION_0kL1OOai9AIeHS7Dm9jfFu.xml]: Description.

File: IFC4x3/PropertyConstants/f/FULLFACE_1gx$Dhph5C_PHaPBNiGN7Y.xml [hunk 11]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="FULLFACE_1gxDhph5C_PHaPBNiGN7Y" Name="FULLFACE" UniqueId="6aeff36b-ceb1-4cf9-9464-64b5ec4171e2" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/f/FULLFACE_1gx$Dhph5C_PHaPBNiGN7Y.xml]: Description.

File: IFC4x3/PropertyConstants/p/PILOTBORE_1k46t6f0P41vmLekOeqpbI.xml [hunk 12]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PILOTBORE_1k46t6f0P41vmLekOeqpbI" Name="PILOTBORE" UniqueId="6e106dc6-a406-4407-9c15-a2e628d33952" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/p/PILOTBORE_1k46t6f0P41vmLekOeqpbI.xml]: Description.

File: IFC4x3/PropertyConstants/r/RAISEDRILL_2N81ytBd90BhBPCttgjznX.xml [hunk 13]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RAISEDRILL_2N81ytBd90BhBPCttgjznX" Name="RAISEDRILL" UniqueId="97201f37-2e72-402e-b2d9-337deab7dc61" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/r/RAISEDRILL_2N81ytBd90BhBPCttgjznX.xml]: Description.

File: IFC4x3/PropertyConstants/r/REAMING_3X1_$v56jCbun7X$BYXNRa.xml [hunk 14]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="REAMING_3X1_v56jCbun7XBYXNRa" Name="REAMING" UniqueId="e107eff9-146b-4c97-8c47-87f2e28576e4" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/r/REAMING_3X1_$v56jCbun7X$BYXNRa.xml]: Description.

File: IFC4x3/PropertyConstants/s/SHAFT_1T82jBUtP2bOup9ZmPM6JR.xml [hunk 15]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="SHAFT_1T82jBUtP2bOup9ZmPM6JR" Name="SHAFT" UniqueId="5d202b4b-7b76-4295-8e33-263c195864db" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/s/SHAFT_1T82jBUtP2bOup9ZmPM6JR.xml]: Description.

File: IFC4x3/PropertyConstants/s/STAGED_1dnkXGXpfCVgPK$VNW68tP.xml [hunk 16]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="STAGED_1dnkXGXpfCVgPKVNW68tP" Name="STAGED" UniqueId="67c6e850-873a-4c7e-a654-fdf5e0188dd9" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/s/STAGED_1dnkXGXpfCVgPK$VNW68tP.xml]: Description.

File: IFC4x3/Properties/e/ExcavationFaceInclinationOther_0vb$DrG6X2cOgI1cvUqXBm/Documentation.md (hunk 9 above)
Comment (larswik) [reply IFC4x3/Properties/e/ExcavationFaceInclinationOther_0vb$DrG6X2cOgI1cvUqXBm/Documentation.md]: Removing this according to agreement.

File: IFC4x3/PropertyConstants/c/CROSSCONNECTION_0kL1OOai9AIeHS7Dm9jfFu.xml (hunk 10 above)
Comment (larswik) [reply IFC4x3/PropertyConstants/c/CROSSCONNECTION_0kL1OOai9AIeHS7Dm9jfFu.xml]: Done!

File: IFC4x3/PropertyConstants/f/FULLFACE_1gx$Dhph5C_PHaPBNiGN7Y.xml (hunk 11 above)
Comment (larswik) [reply IFC4x3/PropertyConstants/f/FULLFACE_1gx$Dhph5C_PHaPBNiGN7Y.xml]: Done!

File: IFC4x3/PropertyConstants/p/PILOTBORE_1k46t6f0P41vmLekOeqpbI.xml (hunk 12 above)
Comment (larswik) [reply IFC4x3/PropertyConstants/p/PILOTBORE_1k46t6f0P41vmLekOeqpbI.xml]: Done!

File: IFC4x3/PropertyConstants/r/RAISEDRILL_2N81ytBd90BhBPCttgjznX.xml (hunk 13 above)
Comment (larswik) [reply IFC4x3/PropertyConstants/r/RAISEDRILL_2N81ytBd90BhBPCttgjznX.xml]: Removed according to input from @jontydunn

File: IFC4x3/PropertyConstants/r/REAMING_3X1_$v56jCbun7X$BYXNRa.xml (hunk 14 above)
Comment (larswik) [reply IFC4x3/PropertyConstants/r/REAMING_3X1_$v56jCbun7X$BYXNRa.xml]: Done!

File: IFC4x3/PropertyConstants/s/SHAFT_1T82jBUtP2bOup9ZmPM6JR.xml (hunk 15 above)
Comment (larswik) [reply IFC4x3/PropertyConstants/s/SHAFT_1T82jBUtP2bOup9ZmPM6JR.xml]: Done!

File: IFC4x3/PropertyConstants/s/STAGED_1dnkXGXpfCVgPK$VNW68tP.xml (hunk 16 above)
Comment (larswik) [reply IFC4x3/PropertyConstants/s/STAGED_1dnkXGXpfCVgPK$VNW68tP.xml]: Done!

**Pull request merged**